app/scripts/controllers/pod.js

```diff
@@ -87,6 +87,67 @@ angular.module('openshiftConsole')
       }
     };
 
+    /**
+     *  Will set the newTerm's isVisible/isUsed to true, while hiding the previous
+     */
+    $scope.onTerminalSelectChange = function(newTerm) {
+      // Make all terminals invisible (Becuase we don't have a pointer to the terminal that is currently visible)
+      _.each($scope.containerTerminals, function(term) {
+        term.isVisible = false;
+      });
+
+      newTerm.isVisible = true;
+      newTerm.isUsed = true;
+    };
+
+    $scope.tabSelect = function() {
+      $scope.terminalTabWasSelected = true;
+    };
+
+    var makeTerminals = function(existingTerminals) {
+      existingTerminals = existingTerminals || [];
+
+      // Remake list with old list's isVisible/isUsed data
+      _.each($scope.pod.spec.containers, function(spec) {
+        var terminal = _.find(existingTerminals, function(term) {
+          return term.containerName === spec.name;
+        });
+
+        if (!terminal) {
+          existingTerminals.push({
+            containerName: spec.name,
+            isVisible: _.get(_.find($scope.containerTerminals, function(term) {
+              return term.containerName === spec.name;
+            }), 'isVisible'),
+            isUsed: _.get(_.find($scope.containerTerminals, function(term) {
+              return term.containerName === spec.name;
+            }), 'isUsed'),
+            containerStatus: _.get(_.find($scope.pod.status.containerStatuses, function(term) {
+              return term.containerName === spec.name;
+            }), 'state')
+          });
+        } else {
+          var containerStatus = _.get(_.find($scope.pod.status.containerStatuses, function(status) {
```

Comment:
Fwiw, shouldn't create a `var` statement inside an `if/else` (in JS).  You might be able to just inline this: `terminal.containerStatus = _.keys(_.get(_.find( /* */ )))`